shuffle/cli/src/shared.rs

```diff
@@ -407,43 +407,52 @@ pub struct NetworksConfig {
 }
 
 impl NetworksConfig {
-    pub fn default() -> Self {
+    pub fn get(&self, network_name: &str) -> Result<Network> {
+        Network::from(self.networks.get(network_name).ok_or_else(|| {
+            anyhow!("Please add specified network to the ~/.shuffle/Networks.json")
+        })?)
+    }
+}
+
+impl Default for NetworksConfig {
+    fn default() -> Self {
         let mut network_map = BTreeMap::new();
-        network_map.insert("localhost".to_string(), Network::default());
+        network_map.insert(LOCALHOST_NAME.to_string(), Network::default());
         NetworksConfig {
             networks: network_map,
         }
     }
-
-    pub fn url_for(&self, network_name: &str) -> Result<Url> {
-        let specified_network = self.networks.get(network_name).ok_or_else(|| {
-            anyhow!("Please add specified network to the ~/.shuffle/Networks.json")
-        })?;
-        Ok(Url::from_str(
-            format!(
-                "{}:{}",
-                specified_network.base, specified_network.dev_api_port
-            )
-            .as_str(),
-        )?)
-    }
 }
 
 #[derive(Debug, Serialize, Deserialize, PartialEq)]
 pub struct Network {
     name: String,
-    base: String,
-    json_rpc_port: u16,
-    dev_api_port: u16,
+    json_rpc_url: Url,
+    dev_api_url: Url,
+    faucet_url: Option<Url>,
 }
 
 impl Network {
-    pub fn default() -> Self {
+    pub fn from(network_ref: &Network) -> Result<Self> {
```

Comment:
This functionality resembles that of `clone()`. Some thoughts:
1. Let's rename this to copy and implement the Clone trait: https://doc.rust-lang.org/std/clone/trait.Clone.html
2. Do you even need this? Can you simply `derive` clone: https://doc.rust-lang.org/std/clone/trait.Clone.html#derivable

Reply:
This isn't a clone though. I'm changing the type between a reference and a value.

Reply:
This is so cool! Using this.